package/src/languageService/kustoLanguageService.ts

```diff
@@ -1451,6 +1452,39 @@ class KustoLanguageService implements LanguageService {
         return Promise.resolve(renderInfo);
     }
 
+    getReferencedTables(document: TextDocument, cursorOffset: number): Promise<{ name: string }[]> {
+        if (!document || !this.isIntellisenseV2()) {
+            return Promise.resolve([]);
+        }
+
+        const script = this.parseDocumentV2(document);
+        let currentBlock = this.getCurrentCommandV2(script, cursorOffset);
+
+        if (!currentBlock) {
+            return Promise.resolve([]);
+        }
+
+        const text = currentBlock.Text;
+
+        const parsedAndAnalyzed = Kusto.Language.KustoCode.ParseAndAnalyze(text, this._kustoJsSchemaV2);
+
+        const tables = this.toArray<sym.TableSymbol>(this._kustoJsSchemaV2.Database.Tables);
+        // We take all referenced symbols in the query
+        const referencedSymbols = this.toArray<Kusto.Language.Syntax.SyntaxNode>(
+            parsedAndAnalyzed.Syntax.GetDescendants(Kusto.Language.Syntax.Expression)
+        ).filter((expression) => expression.ReferencedSymbol !== null)
+            .map((x) => x.ReferencedSymbol) as sym.TableSymbol[];
+
+        // The Intersection between them is the tables that are used in the query.
+        const intersection = referencedSymbols.filter(
+            (referencedSymbol) =>
+                tables.filter((table) => table === referencedSymbol).length > 0
+        );
```

Comment:
this is n*m loop which might be costly (many referenced Symbol * many tables)
convert table list to map and have quick lookup